src/commands/init.ts

```diff
@@ -52,183 +53,138 @@ export default class Init extends Command {
-      cli.action.stop();
+      await action('Creating files', true, copyFiles(files, options)).catch(e => this.error(e, { exit: 2 }));
 
-      this.log('Installing dependencies');
-      await this.installDependencies(flags);
+      const addFrontendScripts: boolean =
+        flags.frontendScripts || (!flags.skipFrontendScripts && (await cli.confirm('Should frontend-related npm scripts for CI/CD be added?')));
+      cli.log('Adding dependencies to package.json...');
+      await modifyPackageJson(projectDir, addFrontendScripts, buildScaffold);
+      await action('Installing dependencies', !verbose, installDependencies(projectDir, verbose)).catch(e =>
+        this.error(`Error during npm install: ${e.message}`, { exit: 2 })
+      );
 
-      cli.action.start('Modify .gitignore');
-      this.modifyGitIgnore(flags);
-      cli.action.stop();
+      cli.log('Modify `.gitignore`...');
+      this.modifyGitIgnore(projectDir);
 
-      this.printSuccessMessage();
+      buildScaffold ? this.printSuccessMessageScaffold() : this.printSuccessMessage();
     } catch (error) {
       this.error(error, { exit: 1 });
     }
   }
 
-  private async determineInitializationType(flags: Flags): Promise<InitType> {
-    if (fs.existsSync(path.resolve(flags.projectDir, 'package.json'))) {
-      return InitType.existingProject;
+  private async shouldBuildScaffold(projectDir: string, buildScaffold: boolean) {
+    if (buildScaffold) {
+      return true;
     }
 
-    this.log('This folder does not contain a `package.json`.');
-    if (flags.initWithExpress || (await cli.confirm('Should a new `express.js` project be initialized in this folder?'))) {
-      return InitType.freshExpress;
+    if (fs.existsSync(path.resolve(projectDir, 'package.json'))) {
+      return false;
     }
 
-    return InitType.existingProject;
-  }
+    this.log('This folder does not contain a `package.json`.');
 
-  private async initProject(flags: Flags, initializationType: InitType): Promise<void> {
-    switch (initializationType) {
-      case InitType.freshExpress:
-        return this.initExpressProject(flags);
-      case InitType.existingProject:
-        return;
-    }
+    return cli.confirm('Should a new `nest.js` project be initialized in this folder?');
   }
 
-  private async initExpressProject(flags: Flags): Promise<void> {
-    cli.action.start('Initializing Express project');
-
-    const params = ['express-generator', '--no-view', '--git'];
-    const dirEmpty = fs.readdirSync(flags.projectDir).length === 0;
-
-    if (!dirEmpty && (flags.force || (await cli.confirm('Directory is not empty. Should the project be initialized anyway?')))) {
-      params.push('--force');
-    }
-    await execa('npx', params, { cwd: flags.projectDir });
-
-    if (!fs.existsSync('.git')) {
-      await execa('git', ['init'], { cwd: flags.projectDir });
+  private async buildScaffold(projectDir: string, { force, verbose }: Flags) {
+    if (fs.readdirSync(projectDir).length !== 0) {
```

Comment:
As this is an async function anyways, why are we using the sync functions?

Reply:
Mostly for consistency and convenience reasons. In some cases (especially checking a file is present) is recommended to use the sync version: https://stackoverflow.com/questions/33969360/difference-between-fs-exists-and-fs-existssync